### ISSUE TITLE
Airship receiver consuming other gcm payloads

### ISSUE THREAD

**Issue by AndroidGecko** [urbanairship/android-library#22]
Hey there

I have two gcm receivers in my app. They are using same SENDER_ID.
Because Airship receiver checks gcm payloads based only on SENDER_ID
```
String sender = gcmIntent.getStringExtra("from");
        if (sender != null && !sender.equals(airship.getAirshipConfigOptions().gcmSender)) {
            Logger.info("Ignoring GCM message from sender: " + sender);
            return;
        }
```
It seems that it is consuming other gcm payloads and adding some analytics for them (verbose logs below for gcm payload from same SENDER_ID but not from Urban Airship)
```
UALib: Analytics - Adding event: push_arrived
UALib: NotificationIdGenerator - Incrementing notification ID count
UALib: NotificationIdGenerator - Notification ID: 1006
```

My question is can I leave it like this? will this mess up my dashboard analytics :( ?
Or is this only some internal analytics which are ignored in dashboard view

Thanks!

**Comment by rlepinski** [comment]
@AndroidGecko That event is currently not exposed in our dashboards so everything should be fine. and the notification ID generator is alright if it keeps incrementing. Ideally you would use 2 different sender IDs for different projects to avoid crossing the streams.

We will address this in the next SDK update by checking for a couple UA keys in the payload as well as the sender ID, but its not a hight priority fix so I don't have a timeline for when it will be patched.

**Comment by AndroidGecko** [comment]
@rlepinski thanks for the info!

**Comment by mangeshsambare** [comment]
Hi, 
I have same issue.  Do you solved this issue?

**Comment by mangeshsambare** [comment]
Hi @rlepinski ,
As @AndroidGecko question, I have same issue. So, I searched about configuring multiple sender id in single app. But android can not configure multiple FCM sender id in single application.  So I want to handle FCM & airship notifications on each separate call.  How can I segregate airship & FCM notification my app? 
Please help me.

**Comment by marc-scig** [comment]
Hi there,

We have a fix in the pipeline. No set date yet but it's looking to be out sometime soon. In the meanwhile as far as I know there's little danger in the SDK consuming these payloads as long as they still reach your custom receiver, so unless you're experiencing other issues we're not aware of I wouldn't worry about it too much for now.

**Comment by mangeshsambare** [comment]
Hi @marc-scig ,
Thanks for replying, We are facing some serious problems related to this. Are you solving this issue within month? Does it in your time-line of one month? . So Kindly ping me when you solve this issue. And Please try to solve this issue as early as possible. 
Thanks!

**Comment by rlepinski** [comment]
@mangeshsambare What are the serious problems you are encountering? We want to make sure it will actually fix your issues.

**Comment by rlepinski** [comment]
Fixed in 8.8.4
